thorlcr/activities/countproject/thcountprojectslave.cpp

```diff
@@ -186,9 +187,9 @@ class CountProjectActivity : public BaseCountProjectActivity, implements ILookAh
         ActivityTimer t(totalCycles, timeActivities);
         if (first) 
         {
-            first = false;
             prevRecCount = count = getPrevCount();
             signalNext();
+            first = false;
```

Comment:
Perhaps not, I was making sure signalNext() is always called so that prevRecCountSem.signal() was called so that prevRecCountSem.wait() would return and onInputFinished() would return.  I didn't know the full extent of which threads call what and offered making first an atomic as an always correct solution.